crates/aranya-client-capi/src/api/defs.rs

```diff
@@ -2228,3 +2236,305 @@ pub unsafe fn aqc_recv_stream_try_recv(
+    control: &[u8],
+    channel: &mut MaybeUninit<AfcChannel>,
+) -> Result<AfcChannelType, imp::Error> {
+    let ctrl = Vec::from(control).into_boxed_slice();
+    let chan = client
+        .rt
+        .block_on(client.inner.afc().recv_ctrl(team_id.into(), ctrl.into()))?;
+
+    let channel_type = match chan {
+        afc::Channel::Bidi(_) => AfcChannelType::Bidirectional,
+        afc::Channel::Send(_) => AfcChannelType::Sender,
+        afc::Channel::Recv(_) => AfcChannelType::Receiver,
+    };
+    AfcChannel::init(channel, chan.into());
+    Ok(channel_type)
+}
+
+/// Returns the [`AfcChannelType`] for a given [`AfcChannel`] container.
+///
+/// @param[in]  channel the AFC channel object [`AfcChannel`].
+/// @param[out] __output the corresponding AFC channel type [`AfcChannelType`].
+#[cfg(feature = "afc")]
+pub fn afc_get_channel_type(channel: &AfcChannel) -> AfcChannelType {
+    match channel.deref().deref() {
+        imp::AfcChannel::Bidi(_) => AfcChannelType::Bidirectional,
+        imp::AfcChannel::Send(_) => AfcChannelType::Sender,
+        imp::AfcChannel::Receive(_) => AfcChannelType::Receiver,
+    }
+}
+
+/// Returns the [`LabelId`] for the associated [`AfcChannel`].
+///
+/// @param[in]  channel the AFC channel object [`AfcChannel`].
+/// @param[out] __output the corresponding label ID [`LabelId`].
+#[cfg(feature = "afc")]
+pub fn afc_get_label_id(channel: &AfcChannel) -> LabelId {
+    match &channel.deref().deref() {
+        imp::AfcChannel::Bidi(c) => c.label_id().into(),
+        imp::AfcChannel::Send(c) => c.label_id().into(),
+        imp::AfcChannel::Receive(c) => c.label_id().into(),
+    }
+}
+
+/// Returns the raw data for a given [`AfcCtrlMsg`].
+///
+/// Note that the lifetime of the pointer is tied to the [`AfcCtrlMsg`].
+///
+/// @param[in]  channel the AFC channel object [`AfcChannel`].
+/// @param[out] ptr the raw pointer of the stored buffer.
+/// @param[out] len the raw length of the stored buffer.
+#[cfg(feature = "afc")]
+pub fn afc_get_msg_data(
+    control: &AfcCtrlMsg,
+    ptr: &mut MaybeUninit<*const u8>,
+    len: &mut MaybeUninit<usize>,
+) {
+    let slice = control.0.as_bytes();
+    ptr.write(slice.as_ptr());
+    len.write(slice.len());
+}
+
+/// Returns the size of the overhead needed for a channel message.
+///
+/// Note that the ciphertext buffer must be at least `plaintext_len` +
+/// `aranya_afc_channel_overhead()` long.
+#[cfg(feature = "afc")]
+pub fn afc_channel_overhead() -> usize {
+    afc::Channels::OVERHEAD
+}
+
+/// Returns the three-way comparison between `seq1` and `seq2`.
+///
+/// @param[in]  seq1 the first sequence number to compare.
+/// @param[in]  seq1 the second sequence number to compare.
+/// @param[out] __output the comparison result (-1 is <, 0 is =, 1 is >).
+#[cfg(feature = "afc")]
+pub fn afc_seq_cmp(seq1: &AfcSeq, seq2: &AfcSeq) -> isize {
+    let seq1 = &seq1.deref().deref().0;
+    let seq2 = &seq2.deref().deref().0;
+    Ord::cmp(seq1, seq2) as isize
```

Comment:
Note you usually don't have to deref manually. If you want to be somewhat explicit here I'd probably do it like this:
```suggestion
    afc::Seq::cmp(&seq1.0, &seq2.0) as isize
```